back/src/Guide/Database/Set.hs

```diff
@@ -0,0 +1,570 @@
+
+
+----------------------------------------------------------------------------
+-- Categories
+----------------------------------------------------------------------------
+
+-- | Set category title.
+setCategoryTitle :: Uid Category -> "title" :! Text -> ExceptT DatabaseError Transaction ()
+setCategoryTitle catId (arg #title -> title) = do
+  let sql = [r|
+        UPDATE categories
+        SET title = $2
+        WHERE uid = $1
+        |]
+      encoder = contrazip2 uidParam textParam
+      decoder = HD.noResult
+  lift $ HT.statement (catId, title) (Statement sql encoder decoder False)
+
+-- | Set category group.
+setCategoryGroup :: Uid Category -> "group_" :! Text -> ExceptT DatabaseError Transaction ()
+setCategoryGroup catId (arg #group_ -> group_) = do
+  let sql = [r|
+        UPDATE categories
+        SET group_ = $2
+        WHERE uid = $1
+        |]
+      encoder = contrazip2 uidParam textParam
+      decoder = HD.noResult
+  lift $ HT.statement (catId, group_) (Statement sql encoder decoder False)
+
+-- | Set category notes.
+setCategoryNotes :: Uid Category -> "notes" :! Text -> ExceptT DatabaseError Transaction ()
+setCategoryNotes catId (arg #notes -> notes) = do
+  let sql = [r|
+        UPDATE categories
+        SET notes = $2
+        WHERE uid = $1
+        |]
+      encoder = contrazip2 uidParam textParam
+      decoder = HD.noResult
+  lift $ HT.statement (catId, notes) (Statement sql encoder decoder False)
+
+-- | Set category status.
+setCategoryStatus :: Uid Category -> CategoryStatus -> ExceptT DatabaseError Transaction ()
+setCategoryStatus catId status = do
+  let sql = [r|
+        UPDATE categories
+        SET status = $2
+        WHERE uid = $1
+        |]
+      encoder = contrazip2 uidParam categoryStatusParam
+      decoder = HD.noResult
+  lift $ HT.statement (catId, status) (Statement sql encoder decoder False)
+
+-- | Set category selections.
+setCategoryEnabledSections
+  :: Uid Category
+  -> "enable" :! Set ItemSection     -- ^ Sections to enable
+  -> "disable" :! Set ItemSection    -- ^ Sections to disable
+  -> ExceptT DatabaseError Transaction ()
+setCategoryEnabledSections catId (arg #enable -> enable) (arg #disable -> disable) = do
+  oldSections <- _categoryEnabledSections <$> getCategory catId
+  let newSections = (oldSections <> enable) Set.\\ disable
+  let sql = [r|
+        UPDATE categories
+        SET enabled_sections = $2
+        WHERE uid = $1
+        |]
+      encoder = contrazip2 uidParam itemSectionSetParam
+      decoder = HD.noResult
+  lift $ HT.statement (catId, newSections) (Statement sql encoder decoder False)
+
+-- | Add itemId to category items_order.
+addItemIdToCategory :: Uid Category -> Uid Item -> ExceptT DatabaseError Transaction ()
+addItemIdToCategory catId itemId = do
+  itemsOrder <- getCategoryItemsOrder catId
+  when (elem itemId itemsOrder) $ throwError ElementIdIsInOrder
+  let addItemId = itemId : itemsOrder
```

Comment:
Currently new items are added to the end, not to the beginning.

This should not change.